src/Unleash/DefaultUnleash.cs

```diff
@@ -235,6 +243,52 @@ private IEnumerable<Constraint> ResolveConstraints(ActivationStrategy activation
             }
         }
 
+        public void ConfigureEvents(Action<EventCallbackConfig> callback)
+        {
+            if (callback == null)
+            {
+                Logger.Error($"UNLEASH: Unleash->ConfigureEvents parameter callback is null");
+                return;
+            }
+
+            try
+            {
+                var evtConfig = new EventCallbackConfig();
+                callback(evtConfig);
+                EventConfig = evtConfig;
+            }
+            catch (Exception ex)
+            {
+                Logger.Error($"UNLEASH: Unleash->ConfigureEvents executing callback threw exception: {ex.Message}");
+            }
+        }
+
+        private void EmitImpressionEvent(string type, UnleashContext context, bool enabled, string name, string variant = null)
+        {
+            if (EventConfig.ImpressionEvent == null)
+            {
+                Logger.Error($"UNLEASH: Unleash->ImpressionData callback is null, unable to emit event");
+                return;
+            }
+
+            try
+            {
+                EventConfig.ImpressionEvent(new ImpressionEvent
+                {
+                    Type = type,
+                    Context = context,
+                    EventId = Guid.NewGuid().ToString(),
+                    Enabled = enabled,
+                    FeatureName = name,
+                    Variant = variant
+                });
+            }
+            catch (Exception ex)
```

Comment:
Same as above, I can't see an obvious exception that could be caught here, is this just being defensive?

Reply:
Here I'm trapping event emitting, if your logger fails for some reason for instance, that won't crash harder than an error in the log

Reply:
Sure, I can live with that, seems polite to log an error if something super unexpected happens here